[EWS][Post-commit] Fix 32-bit argument escaping

justinmichaud opened the pull request:
#### 2459971a5f2580c99d6d55812221fc28314a5ec5
<pre>
[EWS][Post-commit] Fix 32-bit argument escaping
<a href="https://bugs.webkit.org/show_bug.cgi?id=297335">https://bugs.webkit.org/show_bug.cgi?id=297335</a>

Reviewed by Adrian Perez de Castro.

Python was quoting the --cmakeargs=&quot;&quot; argument, causing parsing strangeness.

We also fix compile-jsc-32bit-without-change to actually do the 32-bit
build, and add linux32 before run-javascriptcore-tests.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileJSCOnly32):
(RunJavaScriptCoreTests.run):
* Tools/CISupport/ews-build/steps.py:
(CompileJSC32):
(CompileJSCWithoutChange32):
(RunJavaScriptCoreTests.start):

Canonical link: <a href="https://commits.webkit.org/298756@main">https://commits.webkit.org/298756@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/248789983c59ca821b864731dc7c80f0dea1dc5d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116154 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35815 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26360 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122211 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66711 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118043 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36509 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44403 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88249 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42788 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119102 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104219 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68660 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28246 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22328 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65892 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98519 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22464 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125360 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43048 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32322 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96978 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/115589 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43413 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100408 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96762 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24695 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42037 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19922 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38995 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42935 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42402 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45737 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44106 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

justinmichaud commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -703,6 +703,7 @@ class RunJavaScriptCoreTests(TestWithFailureCount, CustomFlagsMixin, ShellMixin)
     descriptionDone = ["jscore-tests"]
     jsonFileName = "jsc_results.json"
     command = [
+        "linux32" if self.getProperty('architecture') in ["armv7"] else "",
```
Will the empty string break things?

dpino commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -501,7 +501,7 @@ class CompileLLINTCLoop(CompileWebKit):
 class CompileJSCOnly32(CompileWebKit):
     name = 'compile-jsc-32bit'
-    build_command = ["linux32", "perl", "Tools/Scripts/build-jsc", "--32-bit", "--cmakeargs=\"-DUSE_LIBBACKTRACE=OFF -DDEVELOPER_MODE=ON -DENABLE_OFFLINE_ASM_ALT_ENTRY=1 -DCMAKE_CXX_FLAGS='-fuse-ld=gold -Wl,--no-map-whole-files -Wl,--no-keep-memory -Wl,--no-keep-files-mapped -Wl,--no-mmap-output-file -fno-omit-frame-pointer' -DCMAKE_C_FLAGS='-fuse-ld=gold -Wl,--no-map-whole-files -Wl,--no-keep-me ...
+    build_command = ["linux32", "perl", "Tools/Scripts/build-jsc", "--32-bit", "--cmakeargs", "-DUSE_LIBBACKTRACE=OFF -DDEVELOPER_MODE=ON -DENABLE_OFFLINE_ASM_ALT_ENTRY=1 -DCMAKE_CXX_FLAGS='-fuse-ld=gold -Wl,--no-map-whole-files -Wl,--no-keep-memory -Wl,--no-keep-files-mapped -Wl,--no-mmap-output-file -fno-omit-frame-pointer' -DCMAKE_C_FLAGS='-fuse-ld=gold -Wl,--no-map-whole-files -Wl,--no-keep-m ...
```
Another way of passing arguments is to add them into the 'additionalArguments' parameter in `config.json`, for example: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/config.json#L404

I would leave things as they're now, but you may want to try that approach in case this fix doesn't work (for some reason).

Also, when passing arguments to the `build-webkit` command is possible to pass the 'cmakeargs' argument several times. So it would be possible to break down CMake arguments into several parameters. For instance:

"cmakeargs", "foo", "cmakeargs", "bar", "cmakeargs", "quax"...

dpino replied:
@justinmichaud I'm not a reviewer, so you'd need a thumbs up review from an actual reviewer.

justinmichaud committed:
[EWS][Post-commit] Fix 32-bit argument escaping
https://bugs.webkit.org/show_bug.cgi?id=297335

Reviewed by Adrian Perez de Castro.

Python was quoting the --cmakeargs="" argument, causing parsing strangeness.

We also fix compile-jsc-32bit-without-change to actually do the 32-bit
build, and add linux32 before run-javascriptcore-tests.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileJSCOnly32):
(RunJavaScriptCoreTests.run):
* Tools/CISupport/ews-build/steps.py:
(CompileJSC32):
(CompileJSCWithoutChange32):
(RunJavaScriptCoreTests.start):

Canonical link: https://commits.webkit.org/298756@main
